test/components/AppItem.spec.jsx

```diff
@@ -26,6 +26,38 @@ jest.mock('cozy-device-helper', () => ({
   isMobile: jest.fn(),
   openDeeplinkOrRedirect: jest.fn()
 }))
+
+const app = {
+  slug: 'cozy-drive',
+  name: 'Drive',
+  href: 'http://fake.fr'
+}
+
+describe('AppItem', () => {
+  describe('buildAppUrl', () => {
+    const originalBuildQueryParams = AppItem.buildQueryParams
+
+    it('should return untouched href', () => {
+      const wrapper = shallow(<AppItem app={app} t={tMock} />)
+      const url = wrapper.instance().buildAppUrl('http://fake.fr')
+      expect(url.toString()).toBe('http://fake.fr/')
+    })
+
+    it('should update query string', () => {
+      AppItem.buildQueryParams = () => ({
+        foo: 'bar',
+        bar: 'buz'
+      })
+
+      const wrapper = shallow(<AppItem app={app} t={tMock} />)
+      const url = wrapper.instance().buildAppUrl('http://fake.fr')
+      expect(url.toString()).toBe('http://fake.fr/?foo=bar&bar=buz')
```

Comment:
If this fails, buildQueryParams is not put back to the original one. This can lead to false positives for other tests.

One way to mitigate this:

```js
afterEach(() => {
  jest.restoreAllMocks()
})

it('should update query string', () => {
      jest.spyOn(AppItem, 'buildQueryParams').mockImplementation(() => ({
        foo: 'bar',
        bar: 'buz'
      })
     ...rest of the test
})
```

Reply:
Thanks, my mistake, it is how I use to proceed too.